### ISSUE TITLE
Valgrind reports on-exit leak when running rewrite_tag plugin example configuration

### ISSUE THREAD

**Issue by sonofachamp** [fluent/fluent-bit#2066]
## Bug Report

**Describe the bug**
Running the `rewrite_tag` filter without a custom `Emitter_Name` results in a memory leak. It seemed an easy fix would be to call `flb_sds_destroy(ctx->emitter_name)` in the [`cb_rewrite_tag_exit`](https://github.com/fluent/fluent-bit/blob/34cd94f6e24f2a6e3e73f1523088023207be0be1/plugins/filter_rewrite_tag/rewrite_tag.c#L387) function, but that also has an issue when the user does specify a custom `Emitter_Name` since it seems the config map construct wants to be responsible for the memory management in that case.

**To Reproduce**
- Rubular link if applicable:
- Example log message if applicable:
```
==7807== Memcheck, a memory error detector
==7807== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==7807== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==7807== Command: ./bin/fluent-bit -c ../conf/filter_duplicate.conf
==7807== 
Fluent Bit v1.5.0
* Copyright (C) 2019-2020 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2020/03/31 09:11:17] [ info] [storage] version=1.0.3, initializing...
[2020/03/31 09:11:17] [ info] [storage] in-memory
[2020/03/31 09:11:17] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2020/03/31 09:11:17] [ info] [engine] started (pid=7807)
[2020/03/31 09:11:17] [ info] [sp] stream processor started
==7807== Warning: client switching stacks?  SP change: 0x1ffefff388 --> 0x5cb1450
==7807==          to suppress, use: --max-stackframe=137324977976 or greater
==7807== Warning: client switching stacks?  SP change: 0x5cb13c8 --> 0x1ffefff388
==7807==          to suppress, use: --max-stackframe=137324978112 or greater
==7807== Warning: client switching stacks?  SP change: 0x1ffefff388 --> 0x5cb13c8
==7807==          to suppress, use: --max-stackframe=137324978112 or greater
==7807==          further instances of this message will not be shown.
[0] from.test_tag.new.fluent.bit.out: [1585671078.036124905, {"tool"=>"fluent", "sub"=>{"s1"=>{"s2"=>"bit"}}}]
[0] from.test_tag.new.fluent.bit.out: [1585671079.054907669, {"tool"=>"fluent", "sub"=>{"s1"=>{"s2"=>"bit"}}}]
[0] from.test_tag.new.fluent.bit.out: [1585671080.002843091, {"tool"=>"fluent", "sub"=>{"s1"=>{"s2"=>"bit"}}}]
[0] from.test_tag.new.fluent.bit.out: [1585671081.001032331, {"tool"=>"fluent", "sub"=>{"s1"=>{"s2"=>"bit"}}}]
^C[engine] caught signal (SIGINT)
[2020/03/31 09:11:22] [ info] [input] pausing emitter_for_rewrite_tag.0
==7807== 
==7807== HEAP SUMMARY:
==7807==     in use at exit: 81 bytes in 1 blocks
==7807==   total heap usage: 491 allocs, 490 frees, 3,017,530 bytes allocated
==7807== 
==7807== 81 bytes in 1 blocks are definitely lost in loss record 1 of 1
==7807==    at 0x4C2FE96: malloc (vg_replace_malloc.c:309)
==7807==    by 0x158D5B: flb_malloc
==7807==    by 0x158F7A: sds_alloc
==7807==    by 0x1590B6: flb_sds_create_size
==7807==    by 0x1DA777: cb_rewrite_tag_init
==7807==    by 0x15D1FB: flb_filter_init_all
==7807==    by 0x168157: flb_engine_start
==7807==    by 0x14C181: main
==7807== 
==7807== LEAK SUMMARY:
==7807==    definitely lost: 81 bytes in 1 blocks
==7807==    indirectly lost: 0 bytes in 0 blocks
==7807==      possibly lost: 0 bytes in 0 blocks
==7807==    still reachable: 0 bytes in 0 blocks
==7807==         suppressed: 0 bytes in 0 blocks
==7807== 
==7807== For lists of detected and suppressed errors, rerun with: -s
==7807== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
```
- Steps to reproduce the problem:
Run fluent-bit with config:
```
[SERVICE]
    Flush     1
    Log_Level info

[INPUT]
    NAME   dummy
    Dummy  {"tool": "fluent", "sub": {"s1": {"s2": "bit"}}}
    Tag    test_tag

[FILTER]
    Name          rewrite_tag
    Match         test_tag
    Rule          $tool ^(fluent)$  from.$TAG.new.$tool.$sub['s1']['s2'].out false
    # Emitter_Name  

[OUTPUT]
    Name  stdout
    Match from.*
```

**Expected behavior**

No leaks.

**Screenshots**

**Your Environment**

* Version used:
* Configuration:
* Environment name and version (e.g. Kubernetes? What version?):
* Server type and version:
* Operating System and version:
* Filters and plugins:

**Additional context**

I was looking into the `rewrite_tag` filter to see if it could simply duplicate records with a new tag.

**Comment by edsiper** [comment]
thanks for reporting this issue. taking a look now.

**Comment by edsiper** [comment]
type of leak: on exit

**Comment by edsiper** [comment]
Fixed on fe4446bfdc3ed5e40984a87f5d93ad7b918e1ed8, the fix will be part of v1.4.2 release.
